### PR TITLE
cgame: draw small scoreboard when either team exceeds the available rows

### LINKED ISSUE #1435: Toggle small scoreboard for both teams
Currently the scoreboard is toggled to render smaller rows when the amount of player exceeds a certain limit. This, however, only applies to the team whose player count exceeds the limit. The other team is still rendered as usual. This makes for a hard visual comparison between player amounts (and see whether the teams are even).

Instead the scoreboard should render small rows when either team exceeds the limit.

image

### PR DESCRIPTION
**Pull request by timosmit**
This pull request fixes #1435

### CODE REVIEW

**Comment by timosmit** [comment]
Ideally I would like to get rid of both `WM_DrawClientScore` functions as well, but there is quite an amount of inconsistencies in their implementations.

For example these snippets only exist in `WM_DrawClientScore`:

```C
// add some extra space when not showing lives in non-LMS
if (cg_gameType.integer != GT_WOLF_LMS && !livesleft)
{
	maxchars += 2;
}
```

```C
// allow MV clients see the class of its merged client's on the scoreboard
else if (cg.snap->ps.persistant[PERS_TEAM] == ci->team || cg.snap->ps.persistant[PERS_TEAM] == TEAM_SPECTATOR || cgs.clientinfo[cg.clientNum].team == TEAM_SPECTATOR || cg.snap->ps.pm_type == PM_INTERMISSION
#ifdef FEATURE_MULTIVIEW
         || CG_mvMergedClientLocate(score->client)
#endif
```

Are these differences deliberate?

**Comment by rmarquis** [comment]
> Are these differences deliberate?

I'm not sure. Extra space might be required in one version but not the other, since spacing is different due to different font size. I'd however say the multiview has been forgotten.

**Comment by timosmit** [comment]
Good to know. I'll see if I can clean it up a little more.

File: src/cgame/cg_scoreboard.c [hunk 1]
```diff
@@ -1221,11 +1222,13 @@ static int WM_TeamScoreboard(int x, int y, team_t team, float fade, int maxrows,
...
 	}
 
+	row_height = use_mini_chars ? 12 : 16;
+
 	// save off y val
 	tempy = y;
 
 	// draw color bands
-	for (i = 0; i < maxrows; i++)
+    for (i = 0; i < maxrows; i++)
```
Comment (rmarquis) [inline src/cgame/cg_scoreboard.c]: Maybe a small tab/whitespace issue here? Rest of commit LGTM

File: src/cgame/cg_scoreboard.c [hunk 2]
```diff
@@ -1376,9 +1363,20 @@ qboolean CG_DrawScoreboard(void)
 		absmaxrows = 30;
 	}
 
-	WM_TeamScoreboard(x, y, TEAM_AXIS, fade, maxrows, absmaxrows);
+	qboolean players_with_specs_exceed_limit = cg.teamPlayers[TEAM_SPECTATOR] > 0 && (
+			(cg.teamPlayers[TEAM_AXIS] + 1 + (cg.teamPlayers[TEAM_SPECTATOR] + 1) / 2) > maxrows ||
+			(cg.teamPlayers[TEAM_ALLIES] + 1 + (cg.teamPlayers[TEAM_SPECTATOR] + 1) / 2) > maxrows
+	);
+	qboolean players_without_specs_exceed_limit = cg.teamPlayers[TEAM_AXIS] > maxrows || cg.teamPlayers[TEAM_ALLIES] > maxrows;
+	if ((cg.snap->ps.pm_type != PM_INTERMISSION && players_with_specs_exceed_limit) || players_without_specs_exceed_limit)
```
Comment (rmarquis) [inline src/cgame/cg_scoreboard.c]: OCD mode: add an empty line before `if` condition

**Review by rmarquis — COMMENTED** [review]
Very minor nitpick points, but rest LGTM!

**Comment by rmarquis** [comment]
Fixed the minor issues mentioned, and merged. Many thanks for the fix!

**Review by rmarquis — COMMENTED** [review]
I noticed some stylistic issues, so I just ran uncrustify on the file. 
But otherwise LGTM.

**Comment by rmarquis** [comment]
Merged last commit as 0abbd62a17ff495e1f6e7a043e3ae0d48396a407 with mentioned changes.

**Pull request closed without merging**